src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -107,18 +140,41 @@ public ClientBuilder withConnectTimeout(int connectInMillis) {
 	 * 3. currentContext of config file located at ~/.kube/config 
 	 * 
 	 * @return
+	 * @throws KeyManagementException 
 	 */
 	public IClient build() {
 		try {
-			ISSLCertificateCallback sslCallback = defaultIfNull(this.sslCertificateCallback, new NoopSSLCertificateCallback());
+			TrustManagerFactory trustManagerFactory = initTrustManagerFactory(certificateAlias, certificate);
+			X509TrustManager trustManager  = getCurrentTrustManager(trustManagerFactory);
+			SSLContext sslContext = SSLUtils.getSSLContext(trustManager);
+
+			ResponseCodeInterceptor responseCodeInterceptor = new ResponseCodeInterceptor();
+			OpenShiftAuthenticator authenticator = new OpenShiftAuthenticator();
+
+			OkHttpClient.Builder builder = new OkHttpClient.Builder()
+				.addInterceptor(responseCodeInterceptor)
+				.authenticator(authenticator)
+				.readTimeout(readTimeout, readTimeoutUnit)
+				.writeTimeout(writeTimeout, writeTimeoutUnit)
+				.connectTimeout(connectTimeout, connectTimeoutUnit)
+				.hostnameVerifier(this.sslCertificateCallback)
+				.followRedirects(false)
```

Comment:
Re-tested to prove how we this could be made 'true' and updated..  Updated on auth logict to not follow redirects